File: src/main/java/cn/don9cn/blog/action/BaseAction.java
package cn.don9cn.blog.action;

import cn.don9cn.blog.plugins.operaresult.core.OperaResult;

import java.util.List;

/**
 * @Author: liuxindong
 * @Description: 基础action,定义了基础的增删改查
 * @Create: 2017/10/9 15:23
 * @Modify:
 */
public abstract class BaseAction<T> {

    protected abstract OperaResult baseInsert(T t);

    protected abstract OperaResult baseInsertBatch(List<T> list);

    protected abstract OperaResult baseUpdate(T t);

    protected abstract OperaResult baseRemove(String code);

    protected abstract OperaResult baseRemoveBatch(String codes);

    protected abstract OperaResult baseFindById(String code);

    protected abstract OperaResult baseFindAll();

    protected abstract OperaResult baseFindListByParams(T t);

    protected abstract OperaResult baseFindByPage(int page, int limit,String startTime,String endTime, String orderBy, T t);

}

File: src/main/java/cn/don9cn/blog/action/BaseAction.kt
package cn.don9cn.blog.action

import cn.don9cn.blog.support.action.ActionMsg
import cn.don9cn.blog.support.action.ActionMsgHandler

/**
 * 基础action,定义了action层的基础行为
 */
abstract class BaseAction<in T:Any> {

    protected abstract fun baseInsert(t: T): ActionMsg

    protected abstract fun baseInsertBatch(list: List<T>): ActionMsg

    protected abstract fun baseUpdate(t: T): ActionMsg

    protected abstract fun baseRemove(code: String): ActionMsg

    protected abstract fun baseRemoveBatch(codes: String): ActionMsg

    protected abstract fun baseFindById(code: String): ActionMsg

    protected abstract fun baseFindAll(): ActionMsg

    protected abstract fun baseFindListByParams(t: T): ActionMsg

    protected abstract fun baseFindByPage(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: T): ActionMsg

}